tests/nfv/prepare_env.pm

```
@@ -64,22 +73,42 @@ sub run {
+    assert_script_run("curl " . data_url('nfv/conf/02_vswitch.conf') . " -o /root/vswitchperf/conf/02_vswitch.conf");
+    assert_script_run("curl " . data_url('nfv/conf/03_traffic.conf') . " -o /root/vswitchperf/conf/03_traffic.conf");
+    assert_script_run("curl " . data_url('nfv/conf/04_vnf.conf') . " -o /root/vswitchperf/conf/04_vnf.conf");
+    assert_script_run("curl " . data_url('nfv/conf/05_collector.conf') . " -o /root/vswitchperf/conf/05_collector.conf");
+    assert_script_run("curl " . data_url('nfv/conf/06_pktfwd.conf') . " -o /root/vswitchperf/conf/06_pktfwd.conf");
+    assert_script_run("curl " . data_url('nfv/conf/07_loadgen.conf') . " -o /root/vswitchperf/conf/07_loadgen.conf");
+    assert_script_run("curl " . data_url('nfv/conf/08_llcmanagement.conf') . " -o /root/vswitchperf/conf/08_llcmanagement.conf");
+    assert_script_run("curl " . data_url('nfv/conf/10_custom.conf') . " -o /root/vswitchperf/conf/10_custom.conf");
+    assert_script_run("sed -i 's/trafficgen_ip/$trafficgen_ip/' -i /root/vswitchperf/conf/10_custom.conf");
+
+    record_info("Waiting for NFV_TRAFFICGEN_READY");
+    mutex_wait('NFV_TRAFFICGEN_READY', $child_id);
+
+    record_info("Generate ssh keypair and ssh-copy-id to the Traffic generator machine");
     assert_script_run('ssh-keygen -b 2048 -t rsa -q -N "" -f ~/.ssh/id_rsa');
-    exec_and_insert_password("ssh-copy-id -o StrictHostKeyChecking=no root\@$host1");
-    exec_and_insert_password("ssh-copy-id -o StrictHostKeyChecking=no root\@$host2");
+    exec_and_insert_password("ssh-copy-id -o StrictHostKeyChecking=no root\@$trafficgen_ip");
 
-    # Bring mellanox interfaces up
+    record_info("Bring mellanox interfaces up");
     assert_script_run("ip link set dev eth2 up");
     assert_script_run("ip link set dev eth2 up");
 
-    mutex_create("NFV_testing_ready");
-    wait_for_children;
+    record_info("Stop SuSEfirewall2.service");
+    script_run("systemctl stop SuSEfirewall2.service");
```

how about also using `assert_script_run` here? Let's assume this would fail, it could mean that the firewall died unexpectedly and reveal product issues, or not? ;)

---

I wanted that too, but I don't know why, in some cases the command failed and just to be sure I didn't "assert".

---

you explicitly stopping SuSEfirewall2 here,  but test might run at SLE15 too, right?

---

`systemctl 'stop ' . $self->firewall;`  this is one line solution which will work :)

---

Thanks for the hint. Updated.